miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -395,20 +397,19 @@ public JSONObject referenceDataJsonString() throws IOException {
 
     return hot;
   }
-  
+
   @Autowired
   private SampleClassService sampleClassService;
-  
+
   private void populateSampleClasses(ModelMap model) throws IOException {
-    List<SampleClass> sampleClasses = new ArrayList<>(sampleClassService.getAll());
+    List<SampleClass> sampleClasses = new ArrayList<>();
     List<SampleClass> tissueClasses = new ArrayList<>();
-    for (Iterator<SampleClass> i = sampleClasses.iterator(); i.hasNext();) {
-      SampleClass sc = i.next();
+    // Can only create Tissues and Analyte Stock from this page, so remove other classes
+    for (SampleClass sc : sampleClassService.getAll()) {
       if (SampleTissue.CATEGORY_NAME.equals(sc.getSampleCategory())) {
         tissueClasses.add(sc);
-      } else if (!SampleAnalyte.CATEGORY_NAME.equals(sc.getSampleCategory()) || !sc.isStock()) {
-        // Can only create Tissues and Analyte Stock from this page, so remove other classes
-        i.remove();
+      } else if (SampleStock.CATEGORY_NAME.equals(sc.getSampleCategory())) {
+        tissueClasses.add(sc);
```

Comment:
Stock classes shouldn't be added to tissueClasses, but non-stock/tissue classes should be removed from sampleClasses (as was happening before).

sampleClasses = SampleClasses that can be received (created on /sample/new)
tissueClasses = Tissue classes only (used for defining ghost tissue if a stock is being created)